### PR TITLE
MTDSA-34109 Add validation for contractor name in CIS Deductions API

### PR DESCRIPTION
**Pull request by jesse1411**

### CODE REVIEW

File: test/v3/controllers/validators/CreateValidatorFactorySpec.scala [hunk 1]
```diff
@@ -158,6 +159,11 @@ class CreateValidatorFactorySpec extends UnitSpec with MockSharedAppConfig with
         result shouldBe Left(ErrorWrapper(correlationId, RuleTaxYearNotSupportedError))
       }
 
+      "invalid contractor name format is provided" in new Test {
```
Comment (JoNB7777) [inline test/v3/controllers/validators/CreateValidatorFactorySpec.scala:162]: Could you please also add another test for an empty string to check this also triggers the new error? (I checked it manually, and it works fine in bruno, so no issues on that side, but I think it's good practice to have unit tests for both boundaries)

File: it/test/v3/create/CreateControllerISpec.scala [hunk 2]
```diff
@@ -19,27 +19,18 @@ package v3.create
...
-  InternalError,
-  MtdError,
-  NinoFormatError,
-  RuleDateRangeInvalidError,
-  RuleIncorrectOrEmptyBodyError,
-  RuleTaxYearNotEndedError,
-  RuleTaxYearNotSupportedError,
-  ToDateFormatError
-}
+import shared.models.errors.{FromDateFormatError, InternalError, MtdError, NinoFormatError, RuleDateRangeInvalidError, RuleIncorrectOrEmptyBodyError, RuleTaxYearNotEndedError, RuleTaxYearNotSupportedError, ToDateFormatError}
```
Comment (JoNB7777) [inline it/test/v3/create/CreateControllerISpec.scala]: Could you please just replace these with a wildcard import, instead of explicitly importing that many error models

**Review by JoNB7777 — CHANGES_REQUESTED** [review]

File: test/v3/controllers/validators/CreateValidatorFactorySpec.scala (hunk 1 above)
Comment (Taiwo-AB) [reply test/v3/controllers/validators/CreateValidatorFactorySpec.scala:162]: For string values that use the `ResolveStringPattern`, empty string tests will be done once this change to [standardise string validation logic](https://github.com/hmrc/individuals-pensions-income-api/pull/52/changes) is done. So in the validator, a test which doesn't satisfy the regex is fine and in the case upper boundary makes more sense.

**Review by JoNB7777 — APPROVED** [review]
